src/jit/gentree.h

```diff
@@ -4112,6 +4112,7 @@ struct GenTreeSIMD : public GenTreeJitIntrinsic
 struct GenTreeHWIntrinsic : public GenTreeJitIntrinsic
 {
     NamedIntrinsic gtHWIntrinsicId;
+    var_types gtIndexBaseType; // for AVX2 Gather* intrinsics
```

Comment:
Gather intrinsics have complex overloads that need additional information (the base-type of index vector) for codegen, so adding a field in IR. But that let `GenTreeHWIntrinsic` become a large node. @CarolEidt do you think it ok?